### PR TITLE
A basic drive train

### PR DESCRIPTION
**Pull request by DrKittyKat**
Basic drive train set up in "driveTrain.py" component. I believe it should replace "dtFxTest.py", but I left it in for now because I'm not sure.

### CODE REVIEW

File: motorHelper.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        motor.configPeakCurrentDuration(absMaxTimeMs,10)
+        motor.configContinuousCurrentLimit(nominalMaxCurrent,10)
+        motor.enableCurrentLimit(True)
+
+    #if 'rampRate' in motorDescp:
+    #    motor.configOpenLoopRamp(motorDescp['rampRate'],10)
+
+    return motor
+
+class WPI_TalonFeedback(ctre.WPI_TalonSRX):#ctre.wpi_talonsrx.WPI_TalonSRX
```
Comment (tuffinmuffin) [inline motorHelper.py]: docs

Comment (tuffinmuffin) [reply motorHelper.py]: Docs for all classes in here.

File: physics.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#
+# See the notes for the other physics sample
+#
+
+
+from pyfrc.physics import drivetrains
```
Comment (tuffinmuffin) [inline physics.py]: File should be deleted

File: robot.py [hunk 3]
```diff
@@ -6,31 +6,40 @@
 from magicbot import MagicRobot
 
 import components.dtFxTest
```
Comment (tuffinmuffin) [inline robot.py]: Now that we have working drive train, can remove dtFxTest.

File: robot.py [hunk 4]
```diff
@@ -6,31 +6,40 @@
...
         self.right = 0
         self.stick = XboxController(0)
 
+        self.driveTrain_motorsList = dict(self.map.motorsMap.driveMotors)
 
     def teleopPeriodic(self):
         """
         Must include. Called ruing teleop.
         """
         self.left = self.stick.getRawAxis(1)
```
Comment (tuffinmuffin) [inline robot.py]: Move to set drive train input method. Lets keep this function clear and break it up.

File: motorHelper.py (hunk 1 above)
Comment (DrKittyKat) [reply motorHelper.py]: I didn't write motorHelper, but I'll try to document it

**Comment by tuffinmuffin** [comment]
you want to get Caiden to? he has a mr also that needs same addressing

File: motorHelper.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+
+import rev
+import ctre
+
+import logging
+log = logging.getLogger("console") #These logs were set up for testing, should not be persistent, please delete if you see these and I forgot
```
[the remark this answers is not in the document]
Comment (DrKittyKat) [reply motorHelper.py]: I changed some of them to use logging, but I'm not sure if I used the right levels or not. I'm also not sure what to do about getting a better name for it, I can't think of one that we should use instead

**Pull request merged**